src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -2424,6 +2426,9 @@ public void EndConnect(IAsyncResult asyncResult)
             Exception? ex = castedAsyncResult.Result as Exception;
             if (ex != null || (SocketError)castedAsyncResult.ErrorCode != SocketError.Success)
             {
+                SocketsTelemetry.Log.ConnectFailed();
+                SocketsTelemetry.Log.ConnectStop();
```

Comment:
The general pattern is to guard call sites with an IsEnabled check:
```C#
if (SocketsTelemetry.Log.IsEnabled())
{
    SocketsTelemetry.Log.ConnectFailed();
    SocketsTelemetry.Log.ConnectStop();
}
```
That not only helps to minimize overhead when tracing isn't enabled, it helps the linker to strip out all of these methods when the linker's been told to omit EventSource-based tracing from the linked app.

Reply:
Done